src/sources.cpp

```diff
@@ -755,4 +768,29 @@ diffractedplanewave::diffractedplanewave(int g_[3], double axis_[3], std::comple
   p = p_;
 };
 
+/*
+Occasionally, it's useful to check the sum of all the actual source amplitudes
+(e.g. to ensure that the boundary weights are being computed properly).
+`sum_sources()` provides a convenient interface to do just that.
+*/
+std::complex<double> fields::sum_sources(field_type ft) {
+  std::complex<double> total_sources = 0.0;
+
+  for (int i = 0; i < num_chunks; i++)
+    if (chunks[i]->is_mine()) total_sources += chunks[i]->sum_sources(ft);
+
+  return sum_to_all(total_sources);
+}
+
+std::complex<double> fields_chunk::sum_sources(field_type ft) {
```

Comment:
Is this function just used for debugging and was forgotten to be removed from the commit?

Reply:
No I think we should keep it, as mentioned in the code comment above and the blurb above:

> Finally, I added some convenience functions that sum up all the source amplitudes (i.e. perform a sort of "integration") to check if what we're doing is right.